src/netbase.cpp

```diff
@@ -575,8 +575,8 @@ static bool ConnectThroughProxy(const proxyType &proxy, const std::string& strDe
     // do socks negotiation
     if (proxy.randomize_credentials) {
         ProxyCredentials random_auth;
-        random_auth.username = strprintf("%i", insecure_rand());
-        random_auth.password = strprintf("%i", insecure_rand());
+        static std::atomic_int counter;
+        random_auth.username = random_auth.password = strprintf("%i", counter++);
```

Comment:
I don't understand this change. How is this random at all? What was the intent of `proxy.randomize_credentials`?

Reply:
See this comment in the top post of the PR (https://github.com/bitcoin/bitcoin/pull/8914):

> - The use of `insecure_rand` in `ConnectThroughProxy` has been replaced by an atomic integer counter. The only goal here is to have a different credentials pair for each connection to go on a different Tor circuit, it does not need to be random nor unpredictable.